### ISSUE TITLE
Setup CI instance

### ISSUE THREAD

**Issue by limed** [mdn/infra#22]
Need a jenkins server setup on the MDN account, depends on #20 and #19

**Comment by escattone** [comment]
@limed I'm adding the following just so I can also use this issue within the MDN sprint.

# User story
As an MDN developer, I want to ensure our CI code works within IT's CI infrastructure so that we can continue to run tests, send notifications, build and register Docker images, and deploy code.

# Acceptance criteria
* Jenkins service (with admin privileges for @limed, @jwhitlock and @escattone)
* The Kuma `Jenkinsfile` runs successfully for the default case as well as the following custom branches:
    * master
    * stage-push
    * stage-integration-tests
* The Kumascript `Jenkinsfile` runs successfully for the default case as well as the following custom branches:
    * master
    * stage-push
* Notifications successfully sent to IRC #mdndev
* Docker images successfully registered on https://hub.docker.com/
* Successful deployments to new stage instance
* Follow-on tasks are documented

# Tasks
- [ ] Spin-up Jenkins service
- [ ] Configure Jenkins for multi-branch pipelines for Kuma and Kumascript
- [ ] Modify the [Groovy utils in Kuma](https://github.com/mozilla/kuma/blob/master/Jenkinsfiles/utils.groovy) to handle the new multi-branch pipeline names and select the appropriate [region](https://github.com/mozilla/kuma/blob/3453d1a436fa00306972a62a06049ef39ec27aaa/Jenkinsfiles/utils.groovy#L45)
- [ ] Modify [new stage set-up](https://github.com/mdn/infra/blob/master/apps/mdn/mdn-aws/k8s/regions/oregon/stage.sh) to use DockerHub rather than quay.io
- [ ] Test both the Kuma and Kumascript `Jenkinsfile` (as detailed in the acceptance criteria)

**Comment by escattone** [comment]
https://github.com/mdn/kumascript/pull/767

**Comment by escattone** [comment]
- [x] The Kuma `Jenkinsfile` runs successfully for the default case as well as the following custom branches: `master`, `stage-push`, and `stage-integration-tests` (NOTE: the `stage-integration-tests` branch builds with errors, but they are due to the unreliability of the tests)
- [x] The Kumascript Jenkinsfile runs successfully for the default case as well as the following custom branches: `master` and `stage-push`
- [x] Notifications successfully sent to IRC #mdndev
- [x] Docker images successfully registered on https://hub.docker.com/
- [x] Successful deployments of Kuma and Kumascript to new stage instance

**Comment by escattone** [comment]
Within this issue, I didn't include the task of creating and testing a `Jenkinsfile`-based multi-branch pipeline for the https://github.com/mdn/interactive-examples repo. This will be a follow-on task that we will track in a new issue.
